tests/conformance/getBlock_test.go

```diff
@@ -0,0 +1,135 @@
+package tests
+
+import (
+	"context"
+	"encoding/hex"
+	"starknet-p2p-tests/config"
+	"starknet-p2p-tests/protocol/p2p/starknet/spec"
+	synthetic_node "starknet-p2p-tests/tools"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type blockHeaderTestCase struct {
+	name       string
+	startBlock uint64
+	limit      uint64
+	step       uint64
+}
+
+func TestSyntheticNodeMultipleBlockHeadersRequest(t *testing.T) {
+	testCases := []blockHeaderTestCase{
+		{"Basic Consecutive Blocks", 1, 5, 1},
+		{"Non-unit Step", 1, 5, 2},
+		{"Non-standard Start and Limit", 10, 3, 1},
+		{"Large Step", 1, 5, 10},
+		{"High Start Block", 100, 5, 1},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			runBlockHeaderTest(t, tc)
+		})
+	}
+}
+
+func runBlockHeaderTest(t *testing.T, tc blockHeaderTestCase) {
+	ctx, cancel := context.WithTimeout(context.Background(), config.DefaultTestTimeout)
+	defer cancel()
+
+	syntheticNode, err := synthetic_node.New(ctx, t)
+	require.NoError(t, err, "Failed to create synthetic node")
+	require.NoError(t, syntheticNode.Connect(ctx, config.TargetPeerAddress), "Failed to connect to target peer")
+	headers, err := syntheticNode.RequestBlockHeaders(ctx, tc.startBlock, tc.limit, tc.step)
+	require.NoError(t, err, "Failed to request block headers")
+
+	validateBlockHeaders(t, headers, tc)
+}
+
+func validateBlockHeaders(t *testing.T, headers []*spec.BlockHeadersResponse, tc blockHeaderTestCase) {
+	actualHeaders := 0
+	var lastBlockNum uint64
+
+	for i, header := range headers {
+		switch msg := header.HeaderMessage.(type) {
+		case *spec.BlockHeadersResponse_Header:
+			actualHeaders++
+			if msg.Header != nil {
+				validateHeader(t, msg.Header, tc, i)
+				lastBlockNum = msg.Header.Number
+			} else {
+				t.Errorf("Header %d is nil", i)
+			}
+		case *spec.BlockHeadersResponse_Fin:
+			t.Logf("Received Fin message at position %d", i)
+		default:
+			t.Errorf("Unknown message type at position %d: %T", i, msg)
+		}
+	}
+
+	assert.Equal(t, int(tc.limit), actualHeaders, "Expected to receive %d headers", tc.limit)
+	assert.Equal(t, tc.startBlock+(tc.limit-1)*tc.step, lastBlockNum, "Unexpected last block number")
+}
+
+func validateHeader(t *testing.T, header *spec.SignedBlockHeader, tc blockHeaderTestCase, index int) {
+	blockNum := header.Number
+	blockHash := hex.EncodeToString(header.BlockHash.Elements)
+	t.Logf("Header %d: Block Number: %d, BlockHash: 0x%s", index, blockNum, blockHash)
+
+	expectedBlockNum := tc.startBlock + uint64(index)*tc.step
+	assert.Equal(t, expectedBlockNum, blockNum, "Unexpected block number for step %d", tc.step)
+
+	assertValidHeader(t, header)
+}
+
+func assertValidHeader(t *testing.T, header *spec.SignedBlockHeader) {
```

Comment:
Do you think we should add test data, eg store a handful of blocks locally, and use that as the source of truth when running these tests?

Reply:
Yep, that’s exactly what I was thinking, or maybe even fetching the data via RPC and comparing it(?). Let's add this in next PR as this is already big enough.